Wallet vue adapter

ildarH opened the pull request:
A new wallet adapter for Vue 3 that can be easily integrated with both Vue 3 and Nuxt 3 applications.

This PR adds wallet adapter and example for Nuxt 3

ildarH committed:
Add Vue Wallet Adapter

ildarH committed:
Add Nuxt.js example

ildarH committed:
Merge branch 'refs/heads/main' into wallet-vue-adapter

ildarH committed:
update Nuxt.js example

ildarH committed:
Merge branch 'main' into wallet-vue-adapter

ildarH committed:
Merge branch 'refs/heads/main' into wallet-vue-adapter

ildarH committed:
Merge branch 'main' into wallet-vue-adapter

ildarH committed:
Merge branch 'refs/heads/main' into wallet-vue-adapter

ildarH committed:
Merge branch 'refs/heads/main' into wallet-vue-adapter

# Conflicts:
#	pnpm-lock.yaml

0xmaayan reviewed (commented):
Thank you for this amazing contribution! I tried the demo vue dapp locally, but could run it only when I removed mSafe wallet from the [wallets list](https://github.com/aptos-labs/aptos-wallet-adapter/pull/363/files#diff-5b598e2a2febe5ee3f4e0d4322f2e247d4e252d165c034d564befd0e90fd3d65R22) - any idea why?

0xmaayan commented:
> Thank you for this amazing contribution! I tried the demo vue dapp locally, but could run it only when I removed mSafe wallet from the [wallets list](https://github.com/aptos-labs/aptos-wallet-adapter/pull/363/files#diff-5b598e2a2febe5ee3f4e0d4322f2e247d4e252d165c034d564befd0e90fd3d65R22) - any idea why?

Ok, it looks like it is because `@msafe/aptos-wallet-adapter` is not part of the dapp dependencies array. Actually non of the wallet packages are in the dependencies array - why?

ildarH committed:
Merge branch 'refs/heads/main' into wallet-vue-adapter

ildarH committed:
update wallet dependencies

ildarH commented:
> > Thank you for this amazing contribution! I tried the demo vue dapp locally, but could run it only when I removed mSafe wallet from the [wallets list](https://github.com/aptos-labs/aptos-wallet-adapter/pull/363/files#diff-5b598e2a2febe5ee3f4e0d4322f2e247d4e252d165c034d564befd0e90fd3d65R22) - any idea why?
> 
> Ok, it looks like it is because `@msafe/aptos-wallet-adapter` is not part of the dapp dependencies array. Actually non of the wallet packages are in the dependencies array - why?

Hey! I've updated. Sorry for the misunderstanding. Can you check please?

ildarH committed:
Merge branch 'refs/heads/main' into wallet-vue-adapter

# Conflicts:
#	pnpm-lock.yaml

ildarH committed:
Merge branch 'main' into wallet-vue-adapter

0xmaayan commented:
Overall LGTM. Thank you for this contribution!

Some notes:

1. We are missing the transaction hash in the link after successfully submitting a transaction (attached screenshot)
2. Is there a reason you didn't include `Antd` and `MUI` wallet selectors in the demo app?
3. When the wallet network is set to `Mainnet` we should show a warnning message as we have on the nextjs demo app
4. Given this is a community contribution, and to provide the best support, we should lock the [@aptos-labs/wallet-adapter-core version](https://github.com/aptos-labs/aptos-wallet-adapter/pull/363/files#diff-d6f882034fc991cc9d912f2fa1ab77e82ad7c849f6e31818eb0a5b92c93e473aR44) to the latest version this Provider supports. This is to ensure we dont break anything in this package when we do `core` package upgrades 
 
<img width="431" alt="image" src="https://github.com/user-attachments/assets/e20cf603-3b6e-4a79-b2f2-16a3dfa80dcd">

ildarH committed:
Update reviewer
* fix transaction hash in nuxt example
* add mainnet warning alert
* lock wallet-adapter-core version

ildarH commented:
Thank you for your feedback! I've added fixes.
Regarding the second point, I can include some of the libraries, such as Antd (it has vue version for sure) a bit later in a separate pull request.

0xmaayan commented on packages/wallet-adapter-vue/package.json:
```diff
@@ -0,0 +1,59 @@
+    "build": "pnpm build:bundle"
+  },
+  "dependencies": {
+    "@aptos-labs/wallet-adapter-core": "^4.15.0"
```
lets lock it to a specific version
```suggestion
    "@aptos-labs/wallet-adapter-core": "4.15.0"
```

ildarH committed:
Update packages/wallet-adapter-vue/package.json

Co-authored-by: Maayan <maayan@aptoslabs.com>

0xmaayan reviewed (approved):
LGTM!

0xmaayan commented:
@ildarH build is failing, could you run `pnpm install` on the adapter root folder, and then make sure build succeed with `pnpm turbo run build`?
https://github.com/aptos-labs/aptos-wallet-adapter/actions/runs/10563236480/job/29324042334?pr=363

ildarH committed:
Merge branch 'main' into wallet-vue-adapter

# Conflicts:
#	pnpm-lock.yaml

ildarH committed:
add changeset

0xmaayan committed:
Merge branch 'main' into wallet-vue-adapter
